user_sync/config.py

```diff
@@ -457,6 +458,15 @@ def get_rule_options(self):
             default_country_code = directory_config.get_string('default_country_code', True)
             if default_country_code:
                 options['default_country_code'] = default_country_code
+            additional_groups = directory_config.get_list('additional_groups', True) or []
+            additional_groups = [{'source': re.compile(r['source']), 'target': r['target']} for r in additional_groups]
+            options['additional_groups'] = additional_groups
+            sync_options = directory_config.get_dict_config('group_sync_options', True)
+            if sync_options:
+                options['auto_create'] = sync_options.get_bool('auto_create', True)
+        if not new_account_type:
+            new_account_type = user_sync.identity_type.ENTERPRISE_IDENTITY_TYPE
```

Comment:
I'm not sure what is going on here either.  I'm responsible for this change, but I don't remember the rationale behind it.  I agree that the new_account_type check isn't needed, so I'll remove it.

I'll also see if the directory_config check is necessary.  The only reason that variable would be None is if `directory_users` is omitted from the config file.  I would think that the config handler should check for that already so we shouldn't need to double check it here.

Reply:
I'm keeping the `directory_config` check because it's a convenient place to raise an exception if `directory_users` isn't specified.  Otherwise, the error bubbles up from `DictConfig`, which will throw an error when the first expected directory config key isn't found.  Raising the exception here will make it more clear that `directory_users` is missing.  (while I'm at it, I'll do the same thing for `adobe_users`)